src/utils/Api.js

```diff
@@ -0,0 +1,134 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+    .then(res => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+
+  getAppInfo() {
+    // call getUserInfo it in this array
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(res => {
+      if (res.ok) {
+        return res.json();
+      }
+
+      Promise.reject(`Error: ${res.status}`);
+  });
+  // other methods for working with the API
+}
+
+// TODO - implement POST /cards
+addCard({ name, link }) {
+  return fetch(`${this._baseUrl}/cards`, {
+    method: "POST",
+    headers: {
+      ...this._headers,
+    "Content-Type": "application/json",
```

Comment:
Please, use `headers: this._headers,` here. You have `"Content-Type": "application/json"` inside it already